tutorials/flappy-bird/step_8/flappy-bird.cpp

```diff
@@ -0,0 +1,660 @@
+            move_pipe(registry, col.bottom_pipe);
+
+            // If this column is not scored, and player passed this column
+            if (!col.scored && column_pos_x < constants.player_pos_x) {
+                // Increase the score
+                update_score(registry, score_entity);
+
+                // Set column as scored
+                col.scored = true;
+            }
+
+            // If column is out of the screen
+            if (column_pos_x < -constants.column_distance) {
+                // Remove this column
+                col.destroy(registry, entity);
+
+                // Create a new column at far end
+                create_column(registry, furthest_pipe_position(registry) + constants.column_distance);
+            }
+        }
+    }
+
+private:
+    entt::entity score_entity;
+
+    // Find the furthest pipe's position X
+    float furthest_pipe_position(entt::registry &registry) {
+        float furthest = 0.f;
+
+        for (auto entity : registry.view<column>()) {
+            auto &col = registry.get<column>(entity);
+            float x = entity_registry_.get<transform::position_2d>(col.top_pipe.body).x();
+            if (x > furthest) furthest = x;
+        }
+
+        return furthest;
+    }
+
+    // Move the pipe and return the x position
+    float move_pipe(entt::registry &registry, pipe &pipe) {
+        // Retrieve constants
+        const auto constants = registry.ctx<flappy_bird_constants>();
+
+        // Get current position of the pipe
+        auto pos = registry.get<transform::position_2d>(pipe.body);
+
+        // Shift pos X to left by scroll_speed but multiplying with dt because we do this so many times a second,
+        // Delta time makes sure that it's applying over time, so in one second it will move scroll_speed pixels
+        auto new_pos_x = pos.x() - constants.scroll_speed * timer::time_step::get_fixed_delta_time();
+
+        // Set the new position value
+        registry.assign_or_replace<transform::position_2d>(pipe.body, new_pos_x, pos.y());
+
+        // Set cap position too
+        auto cap_pos = registry.get<transform::position_2d>(pipe.cap);
+        registry.assign_or_replace<transform::position_2d>(pipe.cap, new_pos_x, cap_pos.y());
+
+        // Return the info about if this pipe is out of the screen
+        return new_pos_x;
+    }
+};
+
+// Name this system
+REFL_AUTO (type(column_logic));
+
+// Player Logic System
+class player_logic final : public ecs::logic_update_system<player_logic> {
+public:
+    player_logic(entt::registry &registry, entt::entity player_) noexcept : system(registry), player(player_) {}
+
+    // Update, this will be called every tick
+    void update() noexcept final {
+        auto &registry = entity_registry_;
+
+        // Retrieve constants
+        const auto constants = registry.ctx<flappy_bird_constants>();
+
+        // Get current position of the pipe
+        auto pos = registry.get<transform::position_2d>(player);
```

Comment:
Here you are not getting the current position of the pipe but of the player

Reply:
fixed, copy paste issue!